examples/time_frequency/plot_time_frequency_global_field_power.py

```diff
@@ -0,0 +1,141 @@
+
+The objective is to show you how to explore spectrally localized
+effects. For this purpose we adapt the method described in [1] and use it on
+the somato dataset. The idea is to track the band-limited temporal evolution
+of spatial patterns by using the Global Field Power (GFP).
+We first bandpass filter the signals and then apply a Hilbert transform. To
+reveal oscillatory activity the evoked response is then subtracted from every
+single trial. Finally, we rectify the signals prior to averaging across trials.
+Then the GFP is computed as described in [2], using the
+sum of the squares divided by the true degrees of freedom.
+Baselining is subsequently applied to make the GFPs comparable between
+frequencies.
+The procedure is then repeated for each frequency band of interest and
+all GFPs are visualized. To estimate uncertainty, non-parametric confidence
+intervals are computed as described in [3] across channels.
+The advantage of this method over summarizing the Space x Time x Frequency
+output of a Morlet Wavelet in frequency bands is relative speed and, more
+importantly, the clear-cut comparability of the spectral decomposition (the
+same type of filter is used across all bands).
+
+References
+----------
+.. [1] Hari R. and Salmelin R. Human cortical oscillations: a neuromagnetic
+       view through the skull (1997). Trends in Neuroscience 20 (1),
+       pp. 44-49.
+.. [2] Engemann D. and Gramfort A. (2015) Automated model selection in
+       covariance estimation and spatial whitening of MEG and EEG signals,
+       vol. 108, 328-342, NeuroImage.
+.. [3] Efron B. and Hastie T. Computer Age Statistical Inference (2016).
+       Cambrdige University Press, Chapter 11.2.
+"""
+# Authors: Denis A. Engemann <denis.engemann@gmail.com>
+#
+# License: BSD (3-clause)
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+import mne
+from mne.datasets import somato
+
+###############################################################################
+# Set parameters
+data_path = somato.data_path()
+raw_fname = data_path + '/MEG/somato/sef_raw_sss.fif'
+
+# let's explore some frequency bands
+iter_freqs = [
+    ('Theta', 4, 7),
+    ('Alpha', 8, 12),
+    ('Beta', 13, 25),
+    ('Gamma', 30, 45)
+]
+
+###############################################################################
+# We create average power time courses for each frequency band
+
+# set epoching parameters
+event_id, tmin, tmax = 1, -1., 3.
+baseline = None
+
+# get the header to extract events
+raw = mne.io.read_raw_fif(raw_fname, preload=False)
+events = mne.find_events(raw, stim_channel='STI 014')
+
+frequency_map = list()
+
+for band, fmin, fmax in iter_freqs:
+    # (re)load the data to save memory
+    raw = mne.io.read_raw_fif(raw_fname, preload=True)
+    raw.pick_types(meg='grad', eog=True)  # we just look at gradiometers
+
+    # bandpass filter and compute Hilbert
+    raw.filter(fmin, fmax, n_jobs=1)  # use more jobs to speed up.
```

Comment:
these filters will have different transition bandwidths, because `l_trans_bandwidth='auto'` mode makes the bandwidth depend on the freq. To make them more comparable, set all these to the same value (maybe 1?)